### ISSUE TITLE
Rule request: force setTimeout/setInterval return value assignation 

### ISSUE THREAD

**Issue by QuentinFchx** [palantir/tslint#3160]
### Rule request
Forcing users to assign values returned by `setTimeout`, `setInterval` might help avoid memory leaks within applications. _Usually_ you want to clear them at some point (otherwise, just disable the rule).
```js
setTimeout(()=>{ })  // invalid
const timeout = setTimeout(()=>{ })  // valid
```

**Comment by johnwiseheart** [comment]
Hi there,

Before we consider this for the core library, it would be great if you could develop and release the rule as a separate package. You can find instructions on how to do this in our [docs](https://palantir.github.io/tslint/develop/custom-rules/). Going to close this down, but feel free to reopen if you disagree or have comments.

Thanks,
TSLint Team
